src/__tests__/App.test.js

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import { shallow, mount } from 'enzyme';
+import App from '../components/App';
+import configureStore from '../store/rootStore';
+import { BrowserRouter as Router } from 'react-router-dom';
+import { Provider } from 'react-redux';
+
+const APP_VERSION = '0.1.0-SNAPSHOT';
+const FOOTER_VERSION_TEXT = `version: ${APP_VERSION}`;
+
+describe('App', () => {
+    it('shows right version', () => {
+        //  Mock fetch call
+        const mockSuccessResponse = { build: { version: APP_VERSION } };
+        const mockJsonPromise = Promise.resolve(mockSuccessResponse);
+        const mockFetchPromise = Promise.resolve({
+            json: () => mockJsonPromise,
+        });
+        jest.spyOn(global, 'fetch').mockImplementation(() => mockFetchPromise);
```

Comment:
It would be difficult in the future to extend such setup when we will have multiple requests done. So let's simply mock the whole service method call instead. This way it will be easier to provide different responses to different service calls.

Regarding tests folder structure, I propose the following:

```
src
    __tests__
        integration
            about.test.js
            index.js
```

In a test file there should be 2 tests: 1 for successful version load and 1 for failed.

In `index.js` file there will be no actual tests, only utilities to re-use some code pieces.
E.g. you will need  the one to mount the app, one to setup common mocks (e.g. load version will be called in every test since it's a startup call, so you will need to provide a mock for that), common element selectors (e.g. `HEADER.ABOUT`) etc.
Take in mind that the next test you will be writing will be file upload, so your utilities should more or less take that into account.